### DISCUSSION TITLE
Regarding viscoelastic pressure-driven channel flow simulations

### DISCUSSION THREAD

**Discussion by Rishav1991** [fppimenta/rheoTool#66] [Q&A]
Dear Francisco Pimenta

I am trying to run a FENE-P model for viscoelastic flow in a channel flow geometry at Re_b = 5000. The flow is pressure-driven. I want to use periodic boundary conditions at the inlet and outlet of the channel. For the normal OpenFOAM (for the Newtonian case using the PisoFoam solver), I have been using meanVelocityForce to provide an external force to drive the flow (where the Re_b is fixed). Is it possible to do the same for the viscoelastic case in the rheoTool? I could not find any tutorial which uses a similar external force to drive the flow for channel flow. How to implement it for the meanVelocityForce inside rheoFoam tutorial? 

Thanks
Rishav Agrawal

**Comment by fppimenta** [comment]
It should work in the same way, but whether it is suitable to your case or not is up to you. As you know, the above fvOptions introduces a momentum source to get the pre-defined space-averaged U. Only U and diag coeffs of momentum eq are used (the source term is agnostic of the particular fluid rheology).  

You can first try rheoFoam + fvOptions + Newtonian and compare it with pisoFoam to get acquainted.

**Reply by Rishav1991** [reply]
Dear F Pimenta

Thanks for your reply. I will try this as you suggested.

Rishav
